example/service/s3/concatenateObjects/concatenateObjects.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"log"
+	"net/url"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type client struct {
+	s3Client *s3.S3
+	bucket   *string
+}
+
+// concatenate will contenate key1's object to key2's object under the key testKey
+func (c *client) concatenate(key1, key2, key3 string, uploadID *string) (*string, *string, error) {
+	// The first part to be uploaded which is represented as part number 1
+	foo, err := c.s3Client.UploadPartCopy(&s3.UploadPartCopyInput{
+		Bucket:     c.bucket,
+		CopySource: aws.String(url.QueryEscape(*c.bucket + "/" + key1)),
+		PartNumber: aws.Int64(1),
+		Key:        &key3,
+		UploadId:   uploadID,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// The second part that is going to be appended to the newly created testKey
+	// object.
+	bar, err := c.s3Client.UploadPartCopy(&s3.UploadPartCopyInput{
+		Bucket:     c.bucket,
+		CopySource: aws.String(url.QueryEscape(*c.bucket + "/" + key2)),
+		PartNumber: aws.Int64(2),
+		Key:        &key3,
+		UploadId:   uploadID,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	// The ETags are needed to complete the process
+	return foo.CopyPartResult.ETag, bar.CopyPartResult.ETag, nil
+}
+
+func main() {
```

Comment:
would be nice to document the input parameters here also in addition to the readme.